citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/GroupManager.java

```diff
@@ -43,4 +43,10 @@ public ICitationGroup getGroup(IUser user, String groupId) {
         }
         return null;
     }
+    
+    @Override
+    public void deleteLocalGroupCopy(IUser user, String groupId) {
+    	Optional<ICitationGroup> groupOptional = groupRepository.findFirstByGroupId(new Long(groupId));
+    	groupRepository.deleteById(groupOptional.get().getId());
```

Comment:
at the moment there might be several group copies. To make this work, we need to delete all of them. I would add a method to the group repository to delete all groups with a specific group id and then call that. We will have to change that once we have one group per user.